### ISSUE TITLE
Deprecate this package

### ISSUE THREAD

**Issue by remcohaszing** [RickBr0wn/react-uuid#9]
At its core, this package is unrelated to React. It generates a random uuid-like value.

If you need to generate a uuid, they’re better of using the web [`crypto.randomUUID()`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) function. If you need to support Node.js 14, use the [`node:crypto.randomUUID()`](https://nodejs.org/api/crypto.html#cryptorandomuuidoptions). If you need to support even older versions of Node.js, or other runtimes that don’t support the web `crypto.randomUUID()` function, there are better implementations, such as the popular [`uuid`](https://github.com/uuidjs/uuid) package.

However, the both name of this package and the readme suggest that this should be used to generate React keys. This will cause unnecessary rerenders, and negatively impact performance of the application. It’s best to use an uniquely identifying property if possible, such as an id. If this is not available, the next best thing to use is the index.

> **Keys must not change** or that defeats their purpose! Don’t generate them while rendering.

[Rules of Keys](https://beta.reactjs.org/learn/rendering-lists#rules-of-keys)

```jsx
// good
function List({ items }) {
  return (
    <ul>
      {items.map((item) => (
        <li key={item.id}>{item.text}</li>
      ))}
    </ul>
  )
}

// bad
function List({ items }) {
  return (
    <ul>
      {items.map((item, index) => (
        <li key={index}>{item.text}</li>
      ))}
    </ul>
  )
}

// worst
function List({ items }) {
  return (
    <ul>
      {items.map((item) => (
        <li key={crypto.randomUUID()}>{item.text}</li>
      ))}
    </ul>
  )
}
```

I don’t intend to be harsh, but this package appears to exist for the sole purpose of teaching an anti-pattern. Given its popularity, this is harmful to the React community.

**Comment by Clueed** [comment]
Besides not being useful in its intended use case, as stated above, I don't even see how it improves upon  [uuid](https://www.npmjs.com/package/uuid) which works perfectly fine in react.

**Comment by dearlordylord** [comment]
I installed this one thinking it was a hook that would integrate into React lifecycle and looking into the source code I was quite surprised. It *is* misrepresented direly, both in its name and in its documentation [which is an anti-pattern as the topic started stated]

**Comment by bencoullie** [comment]
50 000 weekly downloads and counting. I strongly recommend deprecating this repo - it seems to be doing a lot of damage.

**Comment by RickBr0wn** [comment]
wow. 50k weekly downloads, unbelievable. You are all totally correct, of course. This does not do what I had initially intended it to do. I actually wrote this way back in like 2019, when I was a total newbie trying to self teach myself javascript. I wasn't aware of 'scoped packages' back then, and was just getting into React.

I guess the phrase 'a little knowledge is a dangerous thing' is certainly true here. I have never really looked at this package since writing it, only to push some other peoples contributions. The fact people had contributed led me to think the package must have a use. Reading this thread and looking over the code today, I can obvs see this is terrible, and @remcohaszing is totally correct in suggesting that spreading anti patterns is not good for the community. 

I have now depreciated this package, and I genuinely hope I haven't caused anyone any confusion in learning React. Thanks for your input guys, sorry I missed all the previous messages, I only caught todays.
